app/models/host/managed.rb

```diff
@@ -550,6 +555,19 @@ def self.count_habtm(association)
     association.camelize.constantize.find(counter.keys.compact).map {|i| {:label=>i.to_label, :data =>counter[i.id]}}
   end
 
+  def self.register_provision_method(name, friendly_name)
```

Comment:
Ideally this should be on Foreman::Plugin (app/services/foreman/plugin.rb, test/unit/plugin_test.rb) so it can be added during plugin registration, then inside the list methods you can do something like appending the contents of: `Foreman::Plugin.all.map(&:provisioning_methods)` etc.